controllers/dexclient_controller.go

```diff
@@ -47,13 +52,143 @@ type DexClientReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
 func (r *DexClientReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	// _ = log.FromContext(ctx)
 
-	// your logic here
+	log := ctrllog.FromContext(ctx)
+	dexv1Client := &authv1alpha1.DexClient{}
+	if err := r.Get(ctx, req.NamespacedName, dexv1Client); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
 
+	log.Info("found dexclient", "DexClient.name", dexv1Client.Name, "DexClient.namespace", dexv1Client.Namespace)
+
+	switch {
+
+	// Expect the secret to exist in the current namespace.
+	// The secret has a static name 'mtls-secret-mtls'
+	// If it does not exist, requeue every 2 second
+	// case isNotDefinedMTLSSecretClient(dexv1Client, r, ctx):
+	case isNotDefinedMTLSStruct():
+		log.Info("MTLS values not defined, requeue")
+		return ctrl.Result{RequeueAfter: time.Second}, nil
```

Comment:
Guess we need to introduce something like an exponential backoff policy Similar to [this](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/job_controller.go#L133) but you cannot use that code directly since Kube controller are implemented using informers which are hidden here by kubebuilder/operator-sdk framework.